### PR TITLE
Updated max usage in _merge_dicts for python 3

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
This is a pre-existing issue that QA raised while verifying https://dimagi-dev.atlassian.net/browse/QA-4332

https://sentry.io/organizations/dimagi/issues/2781032554/

All calls to `_merge_dicts` pass in `max` as the resolution, so I just changed that from a param and to an inner function that can handle `None`.

The error is preventing data dictionary syncing but coming out of exports code. I didn't dig into the exact circumstances that cause it or if it could also cause issues in exports. It's in staging sentry ~1500 times but not at all on prod, so I'm guessing it's something edge-case-y on the QA domain where it's manifesting.

## Feature Flag
Data dictionary

### Safety story
Basically depending on automated tests and on this being a straightforward change.

I'll deploy it to staging and verify that the data dictionary syncs for the domain that's having problems.

### Automated test coverage

Not sure. Probably?

### QA Plan

Not requesting QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by esoergel — APPROVED** [review]

**Comment by snopoke** [comment]
> It's in staging sentry ~1500 times but not at all on prod, so I'm guessing it's something edge-case-y on the QA domain where it's manifesting.

Interesting - I don't quite see how this error happens at all either. It requires an app doc to have a null version property.

**Comment by orangejenny** [comment]
> It requires an app doc to have a null version property.

Huh, interesting. It happens for multiple docs in that domain, but I didn't look into how many, or if they were from different apps.

File: corehq/apps/export/models/new.py [hunk 1]
```diff
@@ -2434,6 +2430,11 @@ def _merge_dicts(one, two, resolvefn):
         for key in one.keys() ^ two.keys()
     }
 
+    def resolvefn(a, b):
+        if a is None or b is None:
+            return a or b
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/models/new.py:2435]: That is not the intent. I now realize I'd assumed that `None or 0` would return zero, but you're right. I'll do a followup, thank you!
